tests/plugins/test_userview.py

```diff
@@ -22,10 +22,102 @@ def setUp(self):
 	def test_reset_call(self, reset_mock):
 		client = Client()
 		res = client.put('/user/{}/reset_password/'.format(self.user.pk), json.dumps({
-				'reset_code': self.token,
-				'password': 'foobar',
-			}))
+			'reset_code': self.token,
+			'password': 'foobar',
+		}))
 
 		self.assertEqual(200, res.status_code)
 
 		reset_mock.assert_called_once_with(res.wsgi_request, self.user.pk, self.token, 'foobar')
+
+
+class UserLogic(TestCase):
+
+	def setUp(self):
+		super().setUp()
+		self.user = User.objects.create_user(username='foo', password='bar', is_active=True, is_superuser=True)
+		self.user.save()
+		self.user2 = User.objects.create_user(id=7, username='test', password='user', is_active=False)
+		self.user2.save()
+		self.token = default_token_generator.make_token(self.user2)
+
+	def test_user_login(self):
+		self.client = Client()
+		r = self.client.login(username='foo', password='bar')
+		self.assertTrue(r)
+
+	def test_user_login_wrong_password(self):
+		self.client = Client()
+		r = self.client.login(username='foo', password='wrong_pass')
+		self.assertFalse(r)
+
+	def test_user_activation(self):
+		self.client = Client()
+
+		data = {
+			'activation_code': (str(self.token))
+		}
+
+		r = self.client.put('/user/7/activate/', data=json.dumps(data),
+							content_type='application/json')
+		result = jsonloads(r.content)
+		print(result)
+
+		self.assertEqual(200, r.status_code)
```

Comment:
You should also check the user is activated.

Can you also add tests for the non-happy case? What happens if the user posts a wrong token?